Remove the unity-embed-host

joshpeterson opened the pull request:
This component is no longer necessary, as all of this code has been moved to a repository with the Unity editor/engine code.

This completes https://jira.unity3d.com/browse/VM-2209.

joshpeterson committed:
Remove the unity-embed-host

This component is no longer necessary, as all of this code has been
moved to a repository with the Unity editor/engine code.

This completes https://jira.unity3d.com/browse/VM-2209.

mrvoorhe commented on src/coreclr/vm/mono/mono_coreclr.cpp:
```diff
@@ -1241,29 +1241,6 @@ void list_tpa(const SString& searchPath, SString& tpa)
-extern "C" EXPORT_API void EXPORT_CC mono_unity_initialize_host_apis(initialize_func init_func)
-{
-    HRESULT hr;
-    g_HostStruct = (HostStruct*)malloc(sizeof(HostStruct));
```
Was the declaration of g_HostStruct removed?

joshpeterson replied:
This was removed now at 414354fbcd7633467bad0f55c95947aa687e48fe.

joshpeterson committed:
Remove a define for the emebed host that is not used

joshpeterson committed:
Remove the embedding host struct

This includes the removal of all exported API methods that were
implemented as callbacks into C#.

joshpeterson committed:
Merge remote-tracking branch 'origin/unity-main' into remove-unity-embed-host

joshpeterson committed:
Remove a few things added after merge

During the last merge from unity-main, I incorrectly included some things that were remove in unity-main.

This change removes them again.
